### DISCUSSION TITLE
As shown in the figure below, draw four rectangles, click the mouse event, how to obtain the clicked rectangles, and whether the graphic index can be obtained

### DISCUSSION THREAD

**Discussion by pcbren1** [pixijs/pixijs#7671] [Q&A]
![2021-08-04_234058](https://user-images.githubusercontent.com/86771584/128212606-55c38616-5cc7-4593-a7e1-22500757f1c1.png)

**Comment by ivanpopelyshev** [comment]
No way to get those rects. You have to modify graphicsGeometry `containsPoint` to get them. You looked in that code , right?

**Comment by bigtimebuddy** [comment]
Create four separate Graphics objects instead of drawing them to a single Graphics.

**Reply by pcbren1** [reply]
There are 50K rectangular graphics, and the speed of instantiating n graphics will slow down. Only one graphic can be instantiated with the best efficiency
See if there's any other way？

**Comment by ivanpopelyshev** [comment]
> There are 50K rectangular graphics

rects? use pixi-tilemap, and you even can add scaling there: https://www.html5gamedevs.com/topic/47517-is-there-a-way-to-render-sprites-faster-for-mobile/?tab=comments#comment-260138

Also, in any case, if you have 50k rectangular graphics - **you have to make your own interaction**, usually people do it by modifying pixi interaction. PixiJS does not guarantee fast interaction in > 1k objects, because all algorithms for interaction heavily depend on the case and we cant add all 100 variants of quad-tree in the general purpose open-source project.

**Reply by ivanpopelyshev** [reply]
Honestly, 50k rects on scene , if you dont need to zoom out and see them all, is the clear sign that you failed to make good high-level algorithms that separate them by chunks and show only parts. This is not a task for PixiJS, it has to be coded "above" pixi.
